### PR TITLE
fix(arel): PR 27 — visitor blockers (Fragments space, IN preparable, Intersect/Except parens)

### PR DESCRIPTION
**Pull request by deanmarano**
Part of `docs/arel-alignment-followup-prs.md`, PR 27.

## Changes

Three SQL-correctness fixes in `packages/arel/src/visitors/to-sql.ts`:

**1. Fragments space joiner**
`visitArelNodesFragments` now uses `injectJoin(node.values, " ")` — mirrors Rails' `inject_join o.values, collector, " "`. Previously emitted parts with no separator (`"foobar"`), now correctly emits `"foo bar"`.

**2. `collector.preparable = false` for IN/NOT IN**
Set at the top of the array branch in `visitArelNodesIn`, `visitArelNodesNotIn`, and unconditionally in `visitArelNodesHomogeneousIn` — matches Rails `to_sql.rb:337/592/609`. `Composite` already carried the `preparable` field; this just sets it at the right time.

**3. Intersect/Except hard outer parens**
Switched from recursive `infixValueWithParen` to Rails' `infixValue` + hard `"( "` / `" )"` (Rails `to_sql.rb:216-224`). Nested `Intersect(Intersect(a,b),c)` now renders `( a INTERSECT ( b INTERSECT c ) )` instead of flattening — that's the Rails-correct shape.

Also fixes two pre-existing test regressions on `main`:
- `diff.test.ts` `EXAMPLE` fixture missing `paramSql` (required by schema since #850, test never updated)
- `ar_dump.test.ts` referenced `ar-00` fixture deleted in #897; updated to `ar-06`

## Verification

- `pnpm test` (other project) — all pass
- `pnpm api:compare --package arel` — 884/884 (no regressions)

### CODE REVIEW

File: packages/arel/src/visitors/to-sql.ts [hunk 1]
```diff
@@ -1129,11 +1131,17 @@ export class ToSql extends Visitor implements NodeVisitor<SQLString> {
   }
 
   protected visitArelNodesIntersect(node: Nodes.Intersect): SQLString {
-    return this.infixValueWithParen(node, " INTERSECT ");
+    this.collector.append("( ");
+    this.infixValue(node, " INTERSECT ");
+    this.collector.append(" )");
+    return this.collector;
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/arel/src/visitors/to-sql.ts:1137]: Rails' SQLite visitor does not override `visit_Arel_Nodes_Intersect` or `visit_Arel_Nodes_Except` — confirmed by reading `arel/visitors/sqlite.rb` in the Rails v8.0.2 source. Only `infix_value_with_paren` is overridden there (for UNION/UNION ALL). Since the base `visit_Arel_Nodes_Intersect`/`Except` use `infix_value` (not `infix_value_with_paren`), the SQLite grouping-stripping override never applied to Intersect/Except in Rails either. The updated test expectation (Grouping operands render with parens) is the Rails-correct shape. In practice Rails doesn't produce Grouping-wrapped operands for Intersect/Except — SelectManager#intersect passes AST nodes directly — so this is an edge-case-only behaviour change.

File: packages/arel/src/visitors/sqlite.test.ts [hunk 2]
```diff
@@ -89,13 +89,13 @@ describe("SqliteTest", () => {
     it("INTERSECT strips Grouping wrapped operands", () => {
       const node = new Nodes.Intersect(new Nodes.Grouping(q1().ast), new Nodes.Grouping(q2().ast));
       const sql = new Visitors.SQLite().compile(node);
-      expect(sql).toBe('( SELECT * FROM "users" INTERSECT SELECT * FROM "users" )');
+      expect(sql).toBe('( (SELECT * FROM "users") INTERSECT (SELECT * FROM "users") )');
     });
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/arel/src/visitors/sqlite.test.ts:93]: Same justification as comment 1: Rails' SQLite visitor (`arel/visitors/sqlite.rb`) does not override `visit_Arel_Nodes_Intersect` or `visit_Arel_Nodes_Except`. Only `infix_value_with_paren` is overridden, which applies to UNION/UNION ALL only. The updated expectation (parens around Grouping operands) matches what Rails' SQLite visitor would produce.

File: packages/arel/src/visitors/sqlite.test.ts [hunk 3]
```diff
@@ -89,13 +89,13 @@ describe("SqliteTest", () => {
...
 
     it("EXCEPT strips Grouping wrapped operands", () => {
       const node = new Nodes.Except(new Nodes.Grouping(q1().ast), new Nodes.Grouping(q2().ast));
       const sql = new Visitors.SQLite().compile(node);
-      expect(sql).toBe('( SELECT * FROM "users" EXCEPT SELECT * FROM "users" )');
+      expect(sql).toBe('( (SELECT * FROM "users") EXCEPT (SELECT * FROM "users") )');
     });
```
[the remark this answers is not in the document]
Comment (deanmarano) [reply packages/arel/src/visitors/sqlite.test.ts:99]: Same justification as comment 1: Rails' SQLite visitor (`arel/visitors/sqlite.rb`) does not override `visit_Arel_Nodes_Intersect` or `visit_Arel_Nodes_Except`. Only `infix_value_with_paren` is overridden, which applies to UNION/UNION ALL only. The updated expectation (parens around Grouping operands) matches what Rails' SQLite visitor would produce.
